components/nft-sale-widget/nft-sale-widget.tsx

```diff
@@ -0,0 +1,134 @@
+import { useMemo, useState } from "react";
+
+import {
+  Box,
+  Flex,
+  Image,
+  Link,
+  Switch,
+  Text,
+  useMediaQuery
+} from "@chakra-ui/react";
+import { Window } from "@/components/window";
+
+import { useWindowManager } from "@/components/window-manager";
+
+const SwapWidgetTitle = () => {
+  const [isPrivateSale, setIsPrivateSale] = useState(false);
+
+  // TODO: Hide toggle if user is not whitelisted
+
+  const IS_WHITELISTED = true;
```

Comment:
Good for now. This will be hook. Fetching the merkle root and checking if they are whitelisted.